compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/text/StringHelpers.skiko.kt

```diff
@@ -28,4 +28,157 @@ internal actual fun String.findFollowingBreak(index: Int): Int {
+
+    var currentOffset: Int
+    var nextOffset = charIterator.next()
+    while (nextOffset < offset) {
+        nextOffset = charIterator.next()
+    }
+    currentOffset = nextOffset
+
+    while (nextOffset != BreakIterator.DONE) {
+        nextOffset = charIterator.next()
+        if (currentText.codePointAt(currentOffset).isWhitespace() && !currentText.codePointAt(
+                nextOffset
+            ).isWhitespace()
+        ) {
+            return currentOffset
+        } else {
+            currentOffset = nextOffset
+        }
+    }
+    return currentOffset
+}
+
+/**
+ * Determines whether the character at the specified offset in the string is a whitespace Unicode character.
+ *
+ * @param offset The index of the character to check.
+ * @return `true` if the character at the specified offset is a whitespace character, `false` otherwise.
+ */
+internal fun String.isWhitespace(offset: Int): Boolean {
+    return this.codePointAt(offset).isWhitespace()
+}
+
+/**
+ * Checks if the character at the specified offset in the string is a punctuation Unicode character.
+ *
+ * @param offset The offset of the character to check.
+ * @return true if the character at the specified offset is a punctuation character, false otherwise.
+ */
+internal fun String.isPunctuation(offset: Int): Boolean {
+    return this.codePointAt(offset).isPunctuation()
+}
+
+internal fun String.halfSymbolsOffset(): Int {
+    val symbolsCount = this.codePoints.count()
+    val charIterator = BreakIterator.makeCharacterInstance()
+    charIterator.setText(this)
+    var currentOffset = 0
+    for (i in 0..symbolsCount / 2) {
+        currentOffset = charIterator.next()
+    }
+    return currentOffset
+}
+
+/**
+ * Checks if the given Unicode code point is a whitespace character.
+ *
+ * @return `true` if the code point is a whitespace character, `false` otherwise.
+ */
+private fun CodePoint.isWhitespace(): Boolean {
+    // TODO: Extend this behavior when (if) Unicode will have compound whitespace characters.
+    if (this.charCount() != 1) {
+        return false
+    }
+    return this.toChar().isWhitespace()
+}
+
+/**
+ * Checks if the given Unicode code point is a punctuation character.
+ *
+ * @return 'true' if the CodePoint is a punctuation character, 'false' otherwise.
+ */
+private fun CodePoint.isPunctuation(): Boolean {
+    // TODO: Extend this behavior when (if) Unicode will have compound punctuation characters.
+    if (this.charCount() != 1) {
+        return false
+    }
+    val punctuationSet = setOf(
+        CharCategory.DASH_PUNCTUATION,
+        CharCategory.START_PUNCTUATION,
+        CharCategory.END_PUNCTUATION,
+        CharCategory.CONNECTOR_PUNCTUATION,
+        CharCategory.OTHER_PUNCTUATION,
+        CharCategory.INITIAL_QUOTE_PUNCTUATION,
+        CharCategory.FINAL_QUOTE_PUNCTUATION
+    )
+    punctuationSet.forEach {
+        if (it.contains(this.toChar())) {
+            return true
+        }
+    }
+    return false
```

Comment:
It can be one-liner. Like
```suggestion
    return punctuationSet.any { it.contains(this.toChar()) }
```